app/api/models/domain/TaxYear.scala

```diff
@@ -45,10 +47,14 @@ final case class TaxYear private (private val value: String) {
   val asTysDownstream: String = {
     val year2 = value.toInt - 2000
     val year1 = year2 - 1
-    s"${year1}-$year2"
+    s"$year1-$year2"
   }
 
   override def toString: String = s"TaxYear($value)"
+
+  /** Use this for downstream API endpoints that are known to be TYS.
+    */
+  def useTaxYearSpecificApi(implicit featureSwitches: FeatureSwitches): Boolean = featureSwitches.isTaxYearSpecificApiEnabled && year >= 2024
 }
 
 object TaxYear {
```

Comment:
Could you also bring in the helper functions here, e.g. from self-assessment-bsas-api -

`fromIso()`, `now()`, `isPreviousTaxYear()` and the `writes`

Reply:
("fromDownstreamIntToString" in self-assessment-bsas-api is actually meant to be just fromDownstreamInt, so don't copy that one :slightly_smiling_face:)